test/app/components/old_validations_component.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+# TODO: Remove in v3.0.0
+class OldValidationsComponent < ViewComponent::Base
+  include ActiveModel::Validations
+
+  validates :content, presence: true
+
+  def before_render_check
```

Comment:
should this be `before_render` ?
```suggestion
  def before_render
```

Reply:
I purposefully left this component in place using `before_render_check` to ensure backward compatibility until we delete `before_render_check` ❤️